Missionframework/functions/fn_addActionsPlayer.sqf

```diff
@@ -370,4 +370,20 @@ _player addAction [
     "
 ];
 
+// Drop crate
+_player addAction [
+    "<t color='#FFFF00'>", localize "STR_ACTION_CRATE_DROP", "</t>"] joinString "",
+    {detach ((attachedObjects (_this select 1)) select 0)},
+    nil,
+    -504,
+    true,
+    false,
+    "",
+    "
+        alive _originalTarget
+        && {build_confirmed isEqualTo 0}
+        && {(typeOf ((attachedObjects _target) select 0)) in [KPLIB_b_crateSupply, KPLIB_b_crateAmmo, KPLIB_b_crateFuel]}
+    "
+];
```

Comment:
In general you expect the crate to be always the first attached object. Just thinking about having a IR strobe via ACE attached to you or something similar (backpack on chest, etc.).
Then the crate would be on a different position.
Maybe give it a test with something like
```sqf
!((((attachedObjects _originalTarget) apply {typeOf _x}) arrayIntersect [KPLIB_b_crateSupply, KPLIB_b_crateAmmo, KPLIB_b_crateFuel]) isEqualTo [])
```
Quite messy and may overlook something, but I guess you get what I wanted to point out :)

Reply:
I'll do some tests on it.